redis/src/main/scala/zio/redis/TestExecutor.scala

```diff
@@ -79,21 +86,285 @@ private[redis] final class TestExecutor private (
 
   private[this] def runCommand(name: String, input: Chunk[RespValue.BulkString]): USTM[RespValue] = {
     name match {
-      case api.Connection.Ping.name =>
+      case api.Connection.Auth =>
+        onConnection(name, input)(RespValue.bulkString("OK"))
+
+      case api.Connection.ClientCaching =>
+        clientTrackingInfo.get("flags").flatMap { flagOption =>
+          val flags = flagOption.get
+          if (flags.contains("on")) {
+            if (input.head.asString == "YES" && flags.contains("optin"))
+              clientTrackingInfo
+                .put("flags", flags - "caching-no" + "caching-yes")
+                .map(_ => RespValue.SimpleString("OK"))
+            else if (input.head.asString == "NO" && flags.contains("optout"))
+              clientTrackingInfo
+                .put("flags", flags - "caching-yes" + "caching-no")
+                .map(_ => RespValue.SimpleString("OK"))
+            else
+              STM.succeed {
+                RespValue.Error(
+                  "ERR CLIENT CACHING can be called only when the client is in tracking mode with OPTIN or OPTOUT mode enabled"
+                )
+              }
+          } else
+            STM.succeed {
+              RespValue.Error(
+                "ERR CLIENT CACHING can be called only when the client is in tracking mode with OPTIN or OPTOUT mode enabled"
+              )
+            }
+        }
+
+      case api.Connection.ClientId =>
+        for {
+          id <- clientInfo.get("id")
+        } yield RespValue.Integer(id.get.toLong)
```

Comment:
Simple map is more readable in this case. Also, take care of the unsafe calls you made here (`get`, `toLong`).